File: 0242-valid-anagram/0242-valid-anagram.go
func isAnagram(s string, t string) bool {
    
    if len(s) != len(t) {
        return false
    }
    
    m := make(map[rune]int)
    
    for _, e := range s {
        m[e] += 1
    }
    
    for _, e := range t {
        m[e] -= 1
        if m[e] < 0 {
            return false
        }
    }
    
    return true
}

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        m = [0 for _ in range(26)]
        
        for c in s:
            m[ord(c)-97] += 1
            
        for c in t:
            m[ord(c)-97] -= 1
        
        for e in m:
            if e:
                return False
        
        return True
            